### PR TITLE
test/cql-pytest: GROUP BY with static columns

### PR DESCRIPTION
**Pull request by KrzaQ**
This commit adds a new test case 'test_group_by_static_column_and_tombstones' to verify the behavior of GROUP BY queries with static columns. The test is adapted from Cassandra's test suite and aims to reproduce issue #21267.

Original, larger test:
cassandra_tests/validation/operations/select_group_by_test.py::testGroupByWithPaging()

No backports are necessary, this is a new test.

### CODE REVIEW

File: test/cql-pytest/test_group_by.py [hunk 1]
```diff
@@ -232,5 +232,42 @@ def test_group_by_non_aggregated_mutation_attribute_of_column(cql, table1):
...
+        cql.execute(stmt, [1, 3, 2, 12])
+        cql.execute(stmt, [1, 4, 2, 12])
+        cql.execute(stmt, [1, 4, 3, 6])
+        cql.execute(stmt, [2, 2, 3, 3])
+        cql.execute(stmt, [2, 4, 3, 6])
+        cql.execute(stmt, [4, 8, 2, 12])
+        cql.execute(stmt, [5, 8, 2, 12])
+
+        cql.execute(f'DELETE FROM {table} WHERE a = 1 AND b = 3 AND c = 2;')
+        cql.execute(f'DELETE FROM {table} WHERE a = 5;')
```
Comment (kostja) [inline test/cql-pytest/test_group_by.py:260]: What is the purpose of inserting this row and then deleting it?

File: test/cql-pytest/test_group_by.py [hunk 2]
```diff
@@ -232,5 +232,42 @@ def test_group_by_non_aggregated_mutation_attribute_of_column(cql, table1):
...
+
+        result = cql.execute(f'SELECT a, b, s, count(b), count(s) FROM {table} WHERE a = 3 GROUP BY a')
+        # This is the expected and correct result of the query:
+        assert {(3, None, 3, 0, 1)} == set(result)
+
+        # When we remove the WHERE clause, we get all the rows, and b and count(b)
+        # the wrong values (it seems they have leftover values from the previous row)
+        result = cql.execute(f'SELECT a, b, s, count(b), count(s) FROM {table} GROUP BY a')
+        # FIXME: this test reproduces bug #21267
+        assert {(1, 2, 3, 4, 4), (2, 2, 2, 2, 2), (4, 8, None, 1, 0), (3, None, 3, 0, 1)} == set(result)
```
Comment (kostja) [inline test/cql-pytest/test_group_by.py:270]: This result doesn't seem correct. Shouldn't you assert for a correct result, given the test is already marked xfail anyway?

Comment (kostja) [reply test/cql-pytest/test_group_by.py]: How is "None" for any of the columns a correct result? What is None anyway?

**Review by kostja — APPROVED** [review]

File: test/cql-pytest/test_group_by.py [hunk 3]
```diff
@@ -232,5 +232,37 @@ def test_group_by_non_aggregated_mutation_attribute_of_column(cql, table1):
...
+# This test is from cassandra_tests/validation/operations/select_group_by_test.py::testGroupByWithPaging()
+# Reproduces #21267
+@pytest.mark.xfail(reason="issue #21267")
+def test_group_by_static_column(cql, test_keyspace):
+
+    with new_test_table(cql, test_keyspace, "a int, b int, c int, s int static, d int, primary key (a, b, c)") as table:
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 1;')
+        cql.execute(f'UPDATE {table} SET s = 2 WHERE a = 2;')
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 3;')
+        cql.execute(f'UPDATE {table} SET s = 3 WHERE a = 4;')
```
Comment (nyh) [inline test/cql-pytest/test_group_by.py:244]: Looks good, but personally I would have tried to simplify the test to leave only the minimum required to show the bug. You probably don't need 4 different partitions in this test and perhaps 3 or even 2 would have been enough.

File: test/cql-pytest/test_group_by.py [hunk 4]
```diff
@@ -232,5 +232,37 @@ def test_group_by_non_aggregated_mutation_attribute_of_column(cql, table1):
...
+        cql.execute(stmt, [4, 8, 2, 12])
+
+        result = cql.execute(f'SELECT a, b, s, count(b), count(s) FROM {table} WHERE a = 3 GROUP BY a')
+        # This is the expected and correct result of the query:
+        assert {(3, None, 3, 0, 1)} == set(result)
+
+        # When we remove the WHERE clause, we get all the rows, and b and count(b)
+        # the wrong values (it seems they have leftover values from the previous row)
+        result = cql.execute(f'SELECT a, b, s, count(b), count(s) FROM {table} GROUP BY a')
+        # FIXME: this test reproduces bug #21267
```
Comment (nyh) [inline test/cql-pytest/test_group_by.py:262]: Nitpick: the word "FIXME" here is wrong, it make it sounds like this test needs fixing, and I believe it doesn't - the test is 100% correct, it is the code needs fixing. This test passes on Cassandra, right? (By the way, it's worth mentioning that on the commit message).

**Review by nyh — APPROVED** [review]
Looks good, thanks. Left some trivial comments inline.

**Pull request closed without merging**